R/tm_g_gh_boxplot.R

```diff
@@ -324,13 +316,38 @@ srv_g_boxplot <- function(id,
                           plot_height,
                           plot_width,
                           hline_vars_colors,
-                          hline_vars_labels) {
+                          hline_vars_labels,
+                          module_args) {
   with_reporter <- !missing(reporter) && inherits(reporter, "Reporter")
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
   checkmate::assert_class(data, "reactive")
   checkmate::assert_class(shiny::isolate(data()), "teal_data")
 
   moduleServer(id, function(input, output, session) {
+    output$asdf <- renderUI({
+      resolved_x <- teal.transform::resolve_delayed(
+        module_args$xaxis_var,
+        as.list(data()@env)
+      )
+      resolved_y <- teal.transform::resolve_delayed(
+        module_args$yaxis_var,
+        as.list(data()@env)
+      )
+      resolved_param <- teal.transform::resolve_delayed(
+        module_args$param,
+        as.list(data()@env)
+      )
```

Comment:
This causes reload of the inputs after any change of the filter. It means it will be reverted back-to-defaults each time. In [teal.modules.clinical](https://github.com/insightsengineering/teal.modules.clinical/blob/f84bd96102346beeb144ad6a78ae10799022c0b4/R/tm_t_shift_by_arm.R#L420) `resolve_delayed` is triggered only once - and this is the correct behavior.

```suggestion
      resolved_x <- teal.transform::resolve_delayed(
        module_args$xaxis_var,
        as.list(data()@env)
      )
      resolved_y <- teal.transform::resolve_delayed(
        module_args$yaxis_var,
        as.list(data()@env)
      )
      resolved_param <- teal.transform::resolve_delayed(
        module_args$param,
        as.list(data()@env)
      )
```

Reply:
Makes sense. Isolated it as you suggested.